File: 011. Testing unitario de tu aplicación con Spring Boot, Junit y Mockito/spring-testing-unitario/src/main/java/com/example/controller/BookController.java
package com.example.controller;

import com.example.model.Book;
import com.example.service.BookService;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@Controller
public class BookController {

    private BookService bookService;

    public BookController(BookService bookService) {
        this.bookService = bookService;
    }

    @GetMapping("/books")
    public String findAll(Model model){
        model.addAttribute("books", this.bookService.findAll());
        return "book-list";
    }

    @GetMapping("/books/{id}")
    public String findById(@PathVariable Long id, Model model){
        Book book = this.bookService.findById(id).orElseThrow();
        model.addAttribute("book", book);
        return "book-view";
    }
}

File: 088. Implementa una API Reactiva con Spring WebFlux y R2DBC con Kotlin/spring-webflux-r2dbc-kotlin/src/main/kotlin/com/example/controller/BookController.kt
package com.example.controller

import com.example.service.BookHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType
import org.springframework.web.reactive.function.server.router

@Configuration
class BookController {

    @Bean
    fun mainRouter(handler: BookHandler) = router {
        ("/api/books" and accept(MediaType.APPLICATION_JSON)).nest {
            GET("", handler::findAll)
            GET("/author/{author}", handler::findAllByAuthor)
            GET("/{id}", handler::findById)
            POST("", handler::create)
            PUT("", handler::update)
            DELETE("/{id}", handler::deleteById)
        }
    }
}